packages/frontend/src/components/SetConnectionButton/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { ITestConnectionResult } from '@plumber/types'
+
+import { useCallback, useMemo } from 'react'
+import { Alert, AlertIcon } from '@chakra-ui/react'
+import { Button } from '@opengovsg/design-system-react'
+
+interface SetConnectionButtonProps {
+  onNextStep: () => void
+  readOnly: boolean
+  supportsWebhookRegistration: boolean
+  testResult: ITestConnectionResult | undefined
+  testResultLoading: boolean
+}
+
+const SetConnectionButton = ({
+  onNextStep,
+  readOnly,
+  supportsWebhookRegistration,
+  testResult,
+  testResultLoading,
+}: SetConnectionButtonProps) => {
+  const onSubmit = useCallback(() => {
+    if (
+      supportsWebhookRegistration &&
+      testResult &&
+      !testResult.webhookVerified
+    ) {
+      // do sth
+    } else {
+      onNextStep()
+    }
+  }, [])
+
+  const buttonText = useMemo(() => {
+    if (testResultLoading || !testResult) {
+      return 'Testing connection...'
+    }
+
+    if (!testResult.connectionVerified) {
+      return 'Connection not verified'
+    }
+
+    if (!supportsWebhookRegistration) {
+      return readOnly ? 'Connection verified' : 'Continue'
+    }
+
+    if (!testResult.webhookVerified) {
+      return readOnly ? 'Not connected' : 'Connect'
+    }
+
+    return 'Continue'
+  }, [readOnly, testResultLoading, testResult, supportsWebhookRegistration])
+
+  return (
+    <>
+      {supportsWebhookRegistration && testResult?.message && (
+        <Alert
+          status={testResult?.webhookVerified ? 'success' : 'warning'}
+          borderRadius={4}
+        >
+          <AlertIcon />
+          {testResult.message}
+        </Alert>
+      )}
+      <Button
+        isFullWidth
+        onClick={onSubmit}
+        isDisabled={
+          testResultLoading || !testResult?.connectionVerified || readOnly
```

Comment:
Out of scope: I have a gut feel that we will get some confusion if `connectionVerified` is ever `false`, and people cannot continue (e.g. setting FormSG trigger, and users choses a deleted form is deleted).

Reply:
This should be the expected behaviour right? Users shouldnt be able to continue if their credentials are invalid.

Reply:
Ya agree, thats why out of scope. Just thought showing a more descriptive error instead of a "connection not verified" on the disabled button would be nicer.

But that's a task for U-Bend system.